eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.sun.org.apache.xpath.internal.operations.Bool;
+import javafx.util.Pair;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.Stack;
+
+/**
+ * Check Style Rule: Good Logging Practice
+ *  <ol>
+ *      <li>Non-static Logger instance.</li>
+ *      <li>Logger variable name should be 'logger' in public class and not exist in an implementation package.</li>
+ *      <li>Should not use any external logger class, only use ClientLogger. No slf4j, log4j, or other logging imports are allowed.</li>
+ *      <li>Should refer to the containing class in its constructor:
+ *      public class A {
+ *          private ClientLogger logger = new ClientLogger(A.class);
+ *      }
+ *      </li>
+ *  </ol>
```

Comment:
Even though we aren't generating JavaDoc for the checkstyle rules, you should check how this looks when it is generated and specify it correctly. The code above will not be formatted correctly.